atlasdb-config/src/test/java/com/palantir/atlasdb/factory/timelock/TimestampCorroboratingTimelockServiceTest.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * (c) Copyright 2018 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.factory.timelock;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.Mockito.mock;
+
+import java.util.concurrent.atomic.AtomicLong;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import com.palantir.lock.v2.AutoDelegate_TimelockService;
+import com.palantir.lock.v2.IdentifiedTimeLockRequest;
+import com.palantir.lock.v2.LockImmutableTimestampResponse;
+import com.palantir.lock.v2.StartAtlasDbTransactionResponse;
+import com.palantir.lock.v2.TimelockService;
+import com.palantir.timestamp.TimestampManagementService;
+import com.palantir.timestamp.TimestampRange;
+import com.palantir.timestamp.TimestampService;
+
+public class TimestampCorroboratingTimelockServiceTest {
+    private static final IdentifiedTimeLockRequest IDENTIFIED_TIME_LOCK_REQUEST = IdentifiedTimeLockRequest.create();
+
+    private TimelockService timelockService;
+    private GoBackInTimeTimestampService goBackInTimeTimestampService;
+
+    @Before
+    public void setUp() {
+        goBackInTimeTimestampService = new GoBackInTimeTimestampService();
+        timelockService = TimestampCorroboratingTimelockService.create(
+                getMockTimelockService(goBackInTimeTimestampService));
+    }
+
+    @Test
+    public void getFreshTimestampShouldFail() {
+        getIndividualTimestamps(20);
+        goBackInTimeFor(10);
+
+        assertThatThrownBy(timelockService::getFreshTimestamp)
+                .isInstanceOf(AssertionError.class)
+                .hasMessage(generateClocksWentBackwardsMessage(20, 11));
```

Comment:
let's not do this, just do a `hasMessageContaining`/`hasMessageStartingWith` instead of exact match, I have no idea just looking at this what this message is supposed to be.

Reply:
I prefer to avoid computation in tests, since it's primarily what is read in reviews and I don't want to compute things myself.